docs/admin/upgrading.rst

```diff
@@ -11,7 +11,8 @@ Advice
 
 * **Back up your database** and code
 * Mind any version-specific comments below
-* Upgrade only one minor version at a time (1.6 => 1.7, then 1.7 => 1.8)
+* You can upgrade from any minor version to any higher minor version in the same major (2.0 -> 2.1 or 2.0 -> 2.3)
+* You can only upgrade the latest minor version in the previous major version to any minor version in the next version (2.3 -> 3.0 or 2.3 -> 3.2, but not 2.2 -> 3.x or 2.3 -> 4.x).
```

Comment:
I hope that's not the case. That was the point of asynchronous upgrades and database migrations

Reply:
yes that is the case... maybe you misread... what do you think is incorrect with this upgrade path... or what would you be able to do?

Reply:
I want to upgrade from any version to any future version. It's too much work upgrading local, staging and production installs one release at a time

Reply:
> from any version to any future version

that will probably never happen... we should not commit to a 5 year span of time of work just to be able to save a few hours of upgrading effort... just keep up to date and you will not have the trouble with this

Reply:
it will not be possible to upgrade to 3.x from 2.1 or lower at this moment because of async upgrades in 2.2 that are not present in master

Reply:
We should add them to master.

Reply:
there is really no benefit in trying to be able to accomodate this big upgrade steps. I am going to merge this docs PR as it describes the current situation. If you feel we should improve on this feel free, but i think it will be unmanageable and totally not worth it. Even if core could do it, plugins probably wont be able to make the same jump... leaving you still with the need to make incremental steps

Reply:
Fine, but moving forward there is not reason to require these incremental steps: expose migration API to plugins (in case they have their own tables) and provide elgg-cli method to run batch upgrades in batch. No wonder so many people turned away from Elgg trying to upgrade from 1.8 to 1.12, I did it on 2 sites and I was fed up.

Reply:
Would it make sense to keep/add a remark that the direct upgrade to the latest minor release is only to be done on 2.0 or newer. Imagine someone reading these instructions who wants to update to 3.x while still on 1.8. First thing he might do would be updating directly to 1.12 after reading it not knowing that the instructions were different for older Elgg versions and that update would not work. So, how about something like:

"On Elgg 1.x you can only upgrade one minor version at a time (1.6 => 1.7, then 1.7 => 1.8). On Elgg 2.0 or newer you can upgrade from any minor version to any higher minor version in the same major (2.0 -> 2.1 or 2.0 -> 2.3)."

Reply:
Maybe create a table with source version and steps needed to upgrade to 3.0?